src/routines/write_model.jl

```diff
@@ -1,70 +1,75 @@
+struct OptimizerLog
+
+    termination_status::Enum
+    primal_status::Enum
+    dual_status::Enum
+    solve_time::Float64
+
+end
+
+
 # taking the outputted files for the variable DataFrame and writing them to a featherfile
-function _write_variable_results(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString)
+function write_data(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString; kwargs...)
 
+    file_type = get(kwargs, :file_type, Feather)
     for (k,v) in vars_results
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, vars_results[k])
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, vars_results[k])
     end
 
     return
-
 end
 
-function _write_variable_results(canonical::CanonicalModel, save_path::AbstractString)
 
-    for (k,v) in vars(canonical)
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, _result_dataframe_vars(v))
+function write_data(vars_results::OperationModel, save_path::AbstractString; kwargs...)
+
+    file_type = get(kwargs, :file_type, Feather)
+    for (k,v) in vars(vars_results.canonical)
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, _result_dataframe_vars(v))
+
     end
 
     return
-
 end
 
-function _write_optimizer_log(optimizer_log::Dict{Symbol, Any}, save_path::AbstractString)
-
-    optimizer_log[:termination_status] = Int(optimizer_log[:termination_status])
-    optimizer_log[:primal_status] = Int(optimizer_log[:primal_status])
-    optimizer_log[:dual_status] = Int(optimizer_log[:dual_status])
-    optimizer_log[:solve_time] = optimizer_log[:solve_time]
+function write_data(data::DataFrames.DataFrame, save_path::AbstractString, file_name::String; kwargs...)
 
-    df = DataFrames.DataFrame(optimizer_log)
-    file_path = joinpath(save_path,"optimizer_log.feather")
-    Feather.write(file_path, df)
+    file_type = get(kwargs, :file_type, Feather)
+    file_path = joinpath(save_path,"$file_name.$file_type")
+    file_type.write(file_path, data)
 
     return
-
 end
 
-# taking the outputted files for the time_Series DataFrame and writing them to a featherfile
-function _write_time_stamps(time_stamp::DataFrames.DataFrame, save_path::AbstractString)
+function serialize_opt(optimizer_log::OptimizerLog)
 
-    df = DataFrames.DataFrame(time_stamp)
-    file_path = joinpath(save_path,"time_stamp.feather")
-    Feather.write(file_path, df)
+    return JSON.json(Int(optimizer_log))
```

Comment:
What does this return?

Reply:
A dictionary of the Int within the enum, I was honestly slightly confused by making a JSON script, so I apologize if this wasn't proper

Reply:
I'm a bit confused on what your end goal is.  It looks like you create a DataFrame from this output and then write it to a Feather file.  I'd suggest that writing it to a JSON file would be better.  If that's what you want then here are two things you could try (JSON2 should figure out the types for you):
```
# produce a JSON-encoded string.
text = JSON2.write(optimizer_log)
# Write to a JSON file.
open("out.json, "w") do io
    JSON2.write(io, optimizer_log)
end
```